### PR TITLE
control/controlclient: select newer certificate

### PR DESCRIPTION
**Pull request by sailorfrag**
If multiple certificates match when selecting a certificate, use the one
issued the most recently (as determined by the NotBefore timestamp).
This also adds some tests for the function that performs that
comparison.

Updates tailscale/coral#6

Signed-off-by: Adrian Dewhurst <adrian@tailscale.com>

### CODE REVIEW

File: control/controlclient/sign_supported.go [hunk 1]
```diff
@@ -73,23 +74,46 @@ func isSubjectInChain(subject string, chain []*x509.Certificate) bool {
 	return false
 }
 
-func selectIdentityFromSlice(subject string, ids []certstore.Identity) (certstore.Identity, []*x509.Certificate) {
+func selectIdentityFromSlice(subject string, ids []certstore.Identity, now time.Time) (certstore.Identity, []*x509.Certificate) {
+	bestCandidate := struct {
```
Comment (bradfitz) [inline control/controlclient/sign_supported.go:78]: Less syntaxy:

```go
var bestCandidate struct {
    ...
}
```

File: control/controlclient/sign_supported_test.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
```
Comment (bradfitz) [inline control/controlclient/sign_supported_test.go:1]: wrong year

**Review by bradfitz — APPROVED** [review]

File: control/controlclient/sign_supported.go [hunk 3]
```diff
@@ -73,23 +74,46 @@ func isSubjectInChain(subject string, chain []*x509.Certificate) bool {
...
+		if len(bestCandidate.chain) > 0 && bestCandidate.chain[0].NotBefore.After(chain[0].NotBefore) {
+			continue
+		}
+
+		bestCandidate.id = id
+		bestCandidate.chain = chain
 	}
 
-	return nil, nil
+	return bestCandidate.id, bestCandidate.chain
```
Comment (DentonGentry) [inline control/controlclient/sign_supported.go:116]: If no suitable certificate is found, will bestCandidate.id be nil from the initialization on line 79?

**Review by DentonGentry — APPROVED** [review]

File: control/controlclient/sign_supported.go (hunk 3 above)
Comment (sailorfrag) [reply control/controlclient/sign_supported.go:116]: Yes, they'll be `nil, nil`

Comment (sailorfrag) [reply control/controlclient/sign_supported.go:116]: Also, there is a test case specifically covering this and it verifies that `nil` is returned.
